src/Aspire.Hosting/Dcp/ApplicationExecutor.cs

```diff
@@ -1031,8 +1032,8 @@ private void PrepareProjectExecutables()
                 var launchProfile = project.GetEffectiveLaunchProfile();
```

Comment:
Why can't we move this logic to the project resource? We can detect the env variable there right and apply this logic?

Reply:
What value would that add? Then we need to check the env var in 2 places.

Reply:
It seems like it would be better to co-locate the logic that is dependent on the env var.

Reply:
The resource logic should be as close to the resource as possible. If anything, I'd prefer if we could influence this logic with annotation as much as possible.

Reply:
Spoke to @eerhardt offline and agreed this is a fine fix for GA.